### PR TITLE
Create TR_MethodCallAddress relocation for resolved methods

### PR DESCRIPTION
**Pull request by dchopra001**
Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

File: compiler/z/codegen/S390Instruction.cpp [hunk 1]
```diff
@@ -2477,12 +2477,20 @@ TR::S390RILInstruction::generateBinaryEncoding()
                }
 #endif
             (*(int32_t *) (cursor + 2)) = boi(i2);
+            TR::Symbol *sym = getSymbolReference()->getSymbol();
+            TR::ResolvedMethodSymbol *resolvedMethodSym = sym->getResolvedMethodSymbol();
+            TR_ResolvedMethod *resolvedMethod = resolvedMethodSym ? resolvedMethodSym->getResolvedMethod() : 0;
             if (getSymbolReference() && getSymbolReference()->getSymbol()->castToMethodSymbol()->isHelper())
```
Comment (fjeremic) [inline compiler/z/codegen/S390Instruction.cpp]: The old code used to check if we had a symbol reference before doing something. The new code however assumes we have a symbol reference. Is this correct?

Comment (dchopra001) [reply compiler/z/codegen/S390Instruction.cpp]: No, we should check if the symref exists. I've done some cleanup in this area: d743c95

**Comment by fjeremic** [comment]
@genie-omr build zlinux,zos

**Comment by fjeremic** [comment]
Z only change. Not waiting for Travis.

**Pull request merged**